communicator/ssh/communicator.go

```diff
@@ -273,9 +281,45 @@ func (c *comm) reconnect() (err error) {
 	}
 
 	log.Printf("handshaking with SSH")
-	sshConn, sshChan, req, err := ssh.NewClientConn(c.conn, c.address, c.config.SSHConfig)
+
+	// Default timeout to 1 minute if it wasn't specified (zero value). For
+	// when you need to handshake from low orbit.
+	var duration time.Duration
+	if c.config.HandshakeTimeout == 0 {
+		duration = 1 * time.Minute
+	} else {
+		duration = c.config.HandshakeTimeout
+	}
+
+	timeoutExceeded := time.After(duration)
+	connectionEstablished := make(chan bool, 1)
```

Comment:
If you don't use a value for a channel, and you're only using it to signal, it is more idiomatic to use a `chan struct{}`. It takes up less memory space, but also just signals that the value is useless.